File: backend/issue-service/src/main/java/com/example/issueservice/dto/rabbit/IssueCommentCreatedEvent.java
package com.example.issueservice.dto.rabbit;

import com.example.issueservice.dto.models.IssueComment;

import java.time.Instant;
import java.time.LocalDateTime;
import java.time.ZoneId;

public record IssueCommentCreatedEvent(
        long projectId,
        long issueId,
        long commentId,
        long authorId,
        Instant createdAtUtc
) {
    public static IssueCommentCreatedEvent from(IssueComment comment, long projectId) {
        LocalDateTime commentCreatedAt = comment.getCreatedAt();
        Instant createdAtInstant = (commentCreatedAt != null)
                ? commentCreatedAt.atZone(ZoneId.of("UTC")).toInstant()
                : Instant.now();

        return new IssueCommentCreatedEvent(
                projectId,
                comment.getIssue().getId(),
                comment.getId(),
                comment.getUserId(),
                createdAtInstant
        );
    }
}

File: backend/dashboard-service/Backend.Dashboards.Api/Dto/IssueCommentCreatedEvent.cs
using System.Text.Json.Serialization;

namespace Backend.Shared.DTOs;

public class IssueCommentCreatedEvent
{
    [JsonPropertyName("projectId")]
    public long ProjectId { get; set; }

    [JsonPropertyName("issueId")]
    public long IssueId { get; set; }

    [JsonPropertyName("commentId")]
    public long CommentId { get; set; }

    [JsonPropertyName("authorId")]
    public long AuthorId { get; set; }

    [JsonPropertyName("createdAtUtc")]
    public DateTimeOffset CreatedAtUtc { get; set; } = DateTimeOffset.UtcNow;
}